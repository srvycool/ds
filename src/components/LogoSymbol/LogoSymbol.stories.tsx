import React from 'react';
import { LogoSymbol } from './LogoSymbol';
import { Box } from '../Box/Box';

export default {
  title: 'LogoSymbol',
  component: LogoSymbol,
};

export const Default: React.FC = () => (
  <Box width="300px">
    <LogoSymbol fill="primary500" />
  </Box>
);
