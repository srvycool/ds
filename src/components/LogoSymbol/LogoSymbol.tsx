import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import { Theme } from '../../theme';

export type LogoSymbolProps = React.SVGProps<SVGSVGElement> & {
  fill: keyof Theme['colors'];
};

export const LogoSymbol: React.FC<LogoSymbolProps> = ({ fill, ...props }) => {
  const theme = useContext(ThemeContext);

  return (
    <svg viewBox="0 0 300 300" fill="none" {...props}>
      <path
        d="M261.202 300H38.765c-12.751 0-24.794-6.334-31.878-16.539-7.084-10.557-8.855-23.577-4.605-35.542L85.52 25.169C90.832 11.095 103.23 1.594 118.105.186c14.877-1.408 29.045 5.278 36.837 17.947l139.2 222.75c7.439 11.964 7.793 26.744 1.063 39.06C288.121 292.61 275.37 300 261.202 300z"
        fill={theme.colors[fill]}
      />
    </svg>
  );
};
