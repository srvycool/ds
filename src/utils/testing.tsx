import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
