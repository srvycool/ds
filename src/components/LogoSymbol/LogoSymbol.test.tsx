import React from 'react';
import { render } from '../../utils/testing';
import { LogoSymbol } from './LogoSymbol';

describe('LogoSymbol', () => {
  it('should match snapshot', () => {
    const { container } = render(<LogoSymbol fill="primary500" />);

    expect(container).toMatchSnapshot();
  });
});
