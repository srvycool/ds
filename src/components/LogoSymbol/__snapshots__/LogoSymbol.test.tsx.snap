// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSymbol should match snapshot 1`] = `
<div>
  <svg
    fill="none"
    viewBox="0 0 300 300"
  >
    <path
      d="M261.202 300H38.765c-12.751 0-24.794-6.334-31.878-16.539-7.084-10.557-8.855-23.577-4.605-35.542L85.52 25.169C90.832 11.095 103.23 1.594 118.105.186c14.877-1.408 29.045 5.278 36.837 17.947l139.2 222.75c7.439 11.964 7.793 26.744 1.063 39.06C288.121 292.61 275.37 300 261.202 300z"
      fill="#DA4167"
    />
  </svg>
</div>
`;
